pkg/run/run.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Outreach Corporation. All Rights Reserved.
+
+// Description: Entrypoint to run the app
+
+// Package run provides a function that can be invoked inside of main to set up
+// all the service-standard components we expect every service to run.
+//
+// clients should provide any runners they require as part of the app to the
+// Run function with OptAddRunner, and then call `Run` in their main function.
+package run
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+
+	"github.com/getoutreach/gobox/pkg/app"
+	"github.com/getoutreach/gobox/pkg/async"
+	"github.com/getoutreach/gobox/pkg/env"
+	"github.com/getoutreach/gobox/pkg/olog"
+	"github.com/getoutreach/gobox/pkg/trace"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/automemlimit"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/gomaxprocs"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/shutdown"
+
+	"github.com/getoutreach/httpx/pkg/handlers"
+)
+
+// runOpts are the options to pass to Run to configure your service
+type runOpts struct {
+	httpAppHandler http.Handler
```

Comment:
I think I know why you added this as a static member of the runners, but I'm still not sure it's the right move in the spirit of abstracting our service activity running framework into a library. I think I'd rather just have people to inject this as a generic runner opposed to force this as a default activity.

Reply:
I did it because of metrics, liveness check, readiness check, etc. to a large degree, you don't have a first-class service without this thing

Reply:
Right and that is what I figured, but even as you have it right now it doesn't even provide that. You still have to inject the right one in. That is why I think we should just leverage the generic `runners` field for this too, but I'm ok either way

Reply:
> You still have to inject the right one in.

From what I can tell, it should. My test hits the liveness endpoint... or do you mean something else?

Reply:
Otoh, this prevents adding custom logic to the liveness/healthcheck

Reply:
Talked on slack - this was a misunderstanding on my part